src/ray/raylet/scheduling_queue.cc

```diff
@@ -22,6 +22,40 @@ const std::list<Task> &SchedulingQueue::GetReadyTasks() const {
   return this->ready_tasks_;
 }
 
+ResourceSet SchedulingQueue::GetReadyTaskResources() const {
+  // Iterate over all ready tasks and aggregate total resource demand.
+  ResourceSet ready_task_resources;
+  for (const auto &t : ready_tasks_) {
+    ready_task_resources.OuterJoin(t.GetTaskSpecification().GetRequiredResources());
+  }
+  return ready_task_resources;
+}
+
+ResourceSet SchedulingQueue::GetPlaceableTaskResources() const {
+  ResourceSet scheduled_task_resources;
+  for (const auto &t : placeable_tasks_) {
+    scheduled_task_resources.OuterJoin(t.GetTaskSpecification().GetRequiredResources());
+  }
+  return scheduled_task_resources;
+}
+
+ResourceSet SchedulingQueue::GetRunningTaskResources() const {
+  ResourceSet running_task_resources;
+  for (const auto &t : running_tasks_) {
+    running_task_resources.OuterJoin(t.GetTaskSpecification().GetRequiredResources());
+  }
+  return running_task_resources;
+}
+
+ResourceSet SchedulingQueue::GetLoadTaskResources() const {
+  ResourceSet load_resource_set;
+  load_resource_set.OuterJoin(GetPlaceableTaskResources());
+  load_resource_set.OuterJoin(GetReadyTaskResources());
+  // TODO(atumanov): we may or may not want to include running tasks into load.
```

Comment:
if I understand the question correctly, you are suggesting adding running tasks to the load information (at least in the second part of the question). I may be wrong, but I am not concerned about running tasks, because they count against available resources, which is already accounted for.